### DISCUSSION TITLE
Set SplitView Constraints by absolute number

### DISCUSSION THREAD

**Discussion by richardasmall** [stevengharris/SplitView#18] [Ideas]
Currently it is only possible to set constraints based on fractional sizes of the overall view. Being able to set an absolute size would allow for minimum that can not shrink beyond a certain size requirement.

**Comment by stevengharris** [comment]
I agree this could be useful in some cases. At least in my case, though, it ends up being problematic because there are just too many device variations to account for, and fractions work reasonably well for them. It's probably pretty complicated to sort out, but if you come up with something and/or I can help, let me know.

**Reply by bastiaanterhorst** [reply]
I am needing the same behaviour. One approach that might fit better with your vision for your project might be to allow the constraints to be set dynamically, so that I can calculate the minimum fraction size when the window size changes and update the (for instance) minPFraction. I don't think this is possible today, but I did [see](https://github.com/stevengharris/SplitView/issues/29) that you are planning related changes. Would publishing minPFraction and related values be something you'd consider?

**Reply by bastiaanterhorst** [reply]
Sorry -- realised this is already possible and quite simple: using a @State var to set minPFraction and updating that var after resize to the newly calculated fraction that corresponds to the minimum split view width.

**Comment by stevengharris** [comment]
I moved this from Issues to Discussions. If you (or someone else) wants to work on it, let me know how I can help. I don't have plans to work on this myself.
